linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -815,19 +815,23 @@ private final class Analyzer(config: CommonPhaseConfig,
       b.prependToList(ancestors.filter(_.isInterface))
     }
 
-    private def findProxyCandidates(proxyName: MethodName): List[MethodInfo] = {
-      // This is a manual version of .filter(...).toList, for speed
-      var result: List[MethodInfo] = Nil
+    private def findProxyCandidates(proxyName: MethodName): List[MethodInfo] =
+      proxyCandidates.getOrElse(proxyName, Nil)
+
+    private lazy val proxyCandidates = {
```

Comment:
With this refactoring, is the unrolling of the loop done in https://github.com/scala-js/scala-js/pull/4811/commits/195090eced050090990713d013942a4580d97ebf still justified?